pkg/source/mongodb.go

```diff
@@ -101,3 +115,124 @@ func (m MongoDB) closeSession(session *mgo.Session) error {
 	session.Close()
 	return nil
 }
+
+// it prepares the data to have the transformations in fiddler
+// normalize converts into a map[string]string with the key a breadcrumb to the leaf, and the value being the leaf itself
+func normalizeData(mongoData []map[string]interface{}) ([]map[string]interface{}, error) {
```

Comment:
Consider possibly using https://github.com/stretchr/stew/ instead of flattening all the documents.